pages/thank-you.jsx

```diff
@@ -1,27 +1,40 @@
-// this will ignore that Bad Invocation error which is a bug in JSXHint
-/* jshint -W067 */
-
 import React from 'react';
 import MozillaFooter from '../components/footer-mozilla.jsx';
 import Signup from '../components/signup.jsx';
 import Social from '../components/social.jsx';
 import ThankYouHeader from '../components/thank-you-header.jsx';
-import { IntlMixin } from 'react-intl';
 import analytics from '../assets/js/analytics.js';
 import form from '../scripts/form.js';
+import qs from 'qs';
 
 var ThankYou = React.createClass({
-  mixins: [IntlMixin],
+  contextTypes: {
+    location: React.PropTypes.object,
+    router: React.PropTypes.object,
+    intl: React.PropTypes.object
+  },
+  propTypes: {
+    country: React.PropTypes.string.isRequired,
+    test: React.PropTypes.string,
+    location: React.PropTypes.object
+  },
+  getInitialState: function() {
+    let query = qs.parse(this.props.location.search.replace("?", ""));
```

Comment:
Good question, I wasn't very clear.

It looks like you're parsing a query string, where as before the query string was parsed for us by react router. I know we changed how we're handling react router quite a bit, so that might be the reason for the change? But, I really liked that part about react router.

https://github.com/mozilla/donate.mozilla.org/blob/master/components/client.jsx#L8

That ^^ I liked that. Can that not work anymore?